### PR TITLE
CLM-18: Replace Concept ID with UUID in LabTestAttributeType

### PR DESCRIPTION
**Pull request by tahiraniazi**
**Issues I worked on:**
https://issues.openmrs.org/browse/CLM-18

### CODE REVIEW

File: omod/src/main/webapp/addLabTestAttributeType.jsp [hunk 1]
```diff
@@ -1090,13 +1084,13 @@ legend.scheduler-border {
 	}
 	
 	//CHECK THE CONCEPT EXIST
-	function getCheckExist(conceptId){
-		console.log("Type : "+conceptId);
+	function getCheckExist(conceptUuid){
+		console.log("Concept Uuid: "+conceptUuid);
```
Comment (seekme94) [inline omod/src/main/webapp/addLabTestAttributeType.jsp:1088]: Remove this

File: omod/src/main/webapp/addLabTestAttributeType.jsp [hunk 2]
```diff
@@ -979,15 +979,9 @@ legend.scheduler-border {
...
-	function checkConceptExistent(concepId){
-		console.log("Concept ID : "+concepId);
-		console.log("integer : "+isInt(concepId));
-		if(isInt(concepId)){
-		return	getCheckExist(concepId);
-		}
-		else {
-			return false;
-		}
+	function checkConceptExists(conceptUuid){
```
Comment (seekme94) [inline omod/src/main/webapp/addLabTestAttributeType.jsp:982]: Rename to checkObjectExists(objectClass, objectUuid).
What about datatypes other than concept? All allowed datatypes should be handled.

File: omod/src/main/webapp/addLabTestAttributeType.jsp [hunk 3]
```diff
@@ -798,11 +798,11 @@ legend.scheduler-border {
 				 var dataTypeConfig = document.getElementById('datatypeConfig').value;
 				     if(dataTypeOp == "org.openmrs.customdatatype.datatype.ConceptDatatype"){
 				    	 if(dataTypeConfig != "" && dataTypeConfig != null ){
-				        		if(checkConceptExistent(dataTypeConfig)){
+				        		if(checkConceptExists(dataTypeConfig)){
 				        			 document.getElementById("datatypeconfig").style.display= 'none';
 				        		 }else{	        			 
 				        			  document.getElementById("datatypeconfig").style.display= 'block';	
-				      				  document.getElementById('datatypeconfig').innerHTML = "Concept Id  does not exist in concept dictionary";
+				      				  document.getElementById('datatypeconfig').innerHTML = "Either concept does not exist in concept distionary or please enter valid concept UUID";
```
Comment (seekme94) [inline omod/src/main/webapp/addLabTestAttributeType.jsp:805]: Rephrase to "Unable to get reference object". There are multiple possibilities of error.

**Review by seekme94 — CHANGES_REQUESTED** [review]

File: omod/src/main/webapp/addLabTestAttributeType.jsp (hunk 3 above)
Comment (tahiraniazi) [reply omod/src/main/webapp/addLabTestAttributeType.jsp:805]: There are only two possibilities of this error. And the suggested error message is not user friendly.

File: omod/src/main/webapp/addLabTestAttributeType.jsp (hunk 2 above)
Comment (tahiraniazi) [reply omod/src/main/webapp/addLabTestAttributeType.jsp]: Other datatype don't need datatype config field.

**Pull request merged**